contracts/SnappAuction.sol

```diff
@@ -187,6 +195,51 @@ contract SnappAuction is SnappBase {
         }
     }
 
+    function auctionSolutionBid(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 proposedStateRoot,
+        uint proposedObjectiveValue
+        // bytes32 _orderHash,
+        // bytes32 _standingOrderIndex          // should ensure acting on correct order set
+    ) public {
+        require(
+            slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0,
+            "Previous auction not yet resolved!"
+        );
+        // Verify state root and order hash before continuing.
+        require(coreData.stateRoots[stateIndex()] == _currStateRoot, "Incorrect state root");
+        // TODO - verify acting on correct order hash.
+        // require(
+        //     calculateOrderHash(slot, _standingOrderIndex) == _orderHash,
+        //     "Order hash doesn't agree"
+        // );
+
+        // Ensure that auction batch is inactive, unprocessed and in correct phase for bidding
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");  // This is a bit redundant.
+        require(slot != MAX_UINT && slot <= auctionIndex, "Requested order slot does not exist");
+        require(
+            block.timestamp > auctions[slot].creationTimestamp + 3 minutes ||
+                auctions[slot].size == maxUnreservedOrderCount(),
```

Comment:
closing an auction early because of too many orders will result in a longer bidding period? As in if we close the batch 2 minutes after creation, we will allow a bidding period for 4 minutes?

I think this is fine for now as we will hopefully get rid of the limited order space soon.

Reply:
Yes. this certainly an item of interest! Good find. I had also sketched this up yesterday and meant to include it in the "Problems" section of our daily standup. I don't think its a terrible thing to allow more solving time when the batch is completely full!